File: src/types/rule.ts
import type { Action } from "./action";
import type { ConditionGroup } from "./condition";

export interface Rule {
  id: string;
  folderId: string;
  name: string;
  enabled: boolean;
  stopProcessing: boolean;
  conditions: ConditionGroup;
  actions: Action[];
  position: number;
  createdAt: string;
  updatedAt: string;
}

File: src-tauri/src/models/rule.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

use super::action::Action;
use super::condition::ConditionGroup;

pub type RuleId = String;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Rule {
    pub id: RuleId,
    pub folder_id: String,
    pub name: String,
    pub enabled: bool,
    pub stop_processing: bool,
    pub conditions: ConditionGroup,
    pub actions: Vec<Action>,
    pub position: i32,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
